plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Diagnostics.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE TupleSections         #-}
+module Ide.Plugin.Cabal.Diagnostics
+( errorDiagnostic
+, warningDiagnostic
+  -- * Re-exports
+, FileDiagnostic
+, Diagnostic(..)
+)
+where
+
+import qualified Data.Text              as T
+import           Development.IDE        (FileDiagnostic,
+                                         ShowDiagnostic (ShowDiag))
+import           Distribution.Fields    (showPError, showPWarning)
+import qualified Ide.Plugin.Cabal.Parse as Lib
+import           Language.LSP.Types     (Diagnostic (..),
+                                         DiagnosticSeverity (..),
+                                         DiagnosticSource, NormalizedFilePath,
+                                         Position (Position), Range (Range),
+                                         fromNormalizedFilePath)
+
+-- | Produce a diagnostic from a Cabal parser error
+errorDiagnostic :: NormalizedFilePath -> Lib.PError -> FileDiagnostic
+errorDiagnostic fp err@(Lib.PError pos _) =
+  mkDiag fp (T.pack "parsing") DsError (toBeginningOfNextLine pos) msg
```

Comment:
use `OverloadedStrings`?

Reply:
I think this is not the most informative thing to put in the `source` field. Parsing what? I think this should probably just say "cabal".